tools/e2e-commons/pages/wpcom/authorize.js

```diff
@@ -6,26 +6,20 @@ export default class AuthorizePage extends WpPage {
 		super( page, { expectedSelectors: [ '.jetpack-connect__logged-in-form' ] } );
 	}
 
-	async approve( repeat = true ) {
+	async approve( options ) {
+		const { redirectUrl, repeat = true } = options;
 		const authorizeButtonSelector = '.jetpack-connect__authorize-form button';
 		try {
-			return await Promise.all( [
-				this.click( authorizeButtonSelector ),
-				this.waitToDisappear(),
-				this.waitForDomContentLoaded( 50000 ),
-			] );
```

Comment:
This was a surprising find that all of those promises were fired in parallel which should never be the case because those operations are sequential. It was one of the major reasons for timeouts.